test/gtest/uct/uct_test.cc

```diff
@@ -86,6 +86,54 @@ resource_speed::resource_speed(const uct_worker_h& worker,
 
 const char *uct_test::uct_mem_type_names[] = {"host", "cuda"};
 
+std::vector<uct_test_base::md_resource> uct_test_base::enum_md_resources() {
+
+    static std::vector<uct_test::md_resource> all_md_resources;
+
+    if (all_md_resources.empty()) {
+        uct_component_h *uct_components;
+        unsigned num_components;
+        ucs_status_t status;
+
+        status = uct_query_components(&uct_components, &num_components);
+        ASSERT_UCS_OK(status);
+
+        /* for RAII */
+        ucs::handle<uct_component_h*> cmpt_list(uct_components,
+                                                uct_release_component_list);
+
+        for (unsigned cmpt_index = 0; cmpt_index < num_components; ++cmpt_index) {
+            uct_component_attr_t component_attr = {};
+
+            component_attr.field_mask = UCT_COMPONENT_ATTR_FIELD_NAME |
+                                        UCT_COMPONENT_ATTR_FIELD_MD_RESOURCE_COUNT;
+            /* coverity[var_deref_model] */
+            status = uct_component_query(uct_components[cmpt_index], &component_attr);
+            ASSERT_UCS_OK(status);
+
+            /* Save attributes before asking for MD resource list */
```

Comment:
can be simplified if use `md_rsc.cmpt_attr` (initialize above) and `component_attr` (below) separately.